pym/bob/scm/url.py

```diff
@@ -596,6 +628,9 @@ async def _put(self, invoker, workspaceFile, source, url):
             invoker.fail("Upload not supported for URL scheme: " + url.scheme)
 
     def canSwitch(self, oldScm):
+        if self.__separateDownload and not self.__version.get("valid"):
```

Comment:
I think the proper comparison is

```python
if self.__separateDownload != oldScm.__separateDownload:
    return False
```

Basically, this should make whole `VERSION` mechanism obsolete, does it?